src/ray/raylet/agent_manager.cc

```diff
@@ -102,35 +104,11 @@ void AgentManager::StartAgent() {
     RAY_LOG(WARNING) << "Agent process with pid " << child.GetId()
                      << " exit, return value " << exit_code << ". ip "
                      << agent_ip_address_ << ". pid " << agent_pid_;
-    if (agent_restart_count_ < RayConfig::instance().agent_max_restart_count()) {
-      RAY_UNUSED(delay_executor_(
-          [this] {
-            agent_restart_count_++;
-            StartAgent();
-          },
-          // Retrying with exponential backoff
-          RayConfig::instance().agent_restart_interval_ms() *
-              std::pow(2, (agent_restart_count_ + 1))));
-    } else {
-      RAY_LOG(WARNING) << "Agent has failed to restart for "
-                       << RayConfig::instance().agent_max_restart_count()
-                       << " times in a row without registering the agent. This is highly "
-                          "likely there's a bug in the dashboard agent. Please check out "
-                          "the dashboard_agent.log file.";
-      RAY_EVENT(WARNING, EL_RAY_AGENT_EXIT)
-              .WithField("ip", agent_ip_address_)
-              .WithField("pid", agent_pid_)
-          << "Agent failed to be restarted "
-          << RayConfig::instance().agent_max_restart_count()
-          << " times. Agent won't be restarted.";
-      if (RayConfig::instance().raylet_shares_fate_with_agent()) {
-        RAY_LOG(ERROR) << "Raylet exits immediately because the ray agent has failed. "
-                          "Raylet fate shares with the agent. It can happen because the "
-                          "Ray agent is unexpectedly killed or failed. See "
-                          "`dashboard_agent.log` for the root cause.";
-        QuickExit();
-      }
-    }
+    RAY_LOG(ERROR) << "Raylet exits immediately because the ray agent has failed. "
+                      "Raylet fate shares with the agent. It can happen because the "
+                      "Ray agent is unexpectedly killed or failed. See "
+                      "`dashboard_agent.log` for the root cause.";
```

Comment:
```suggestion
    RAY_LOG(ERROR) << "The raylet exited immediately because the Ray agent failed. "
                      "The raylet fate shares with the agent. This can happen because the "
                      "Ray agent was unexpectedly killed or failed. See "
                      "`dashboard_agent.log` for the root cause.";
```